### PR TITLE
Mark timestamp query inside encoders/passes tests as flaky

### PR DESCRIPTION
**Pull request by teoxoy**
We keep seeing these tests failing intermittently in CI, mark them as flaky since https://github.com/gfx-rs/wgpu/issues/2521 already tracks this issue.

### CODE REVIEW

File: examples/src/timestamp_queries/mod.rs [hunk 1]
```diff
@@ -456,7 +456,9 @@ mod tests {
                 .features(
                     wgpu::Features::TIMESTAMP_QUERY
                         | wgpu::Features::TIMESTAMP_QUERY_INSIDE_ENCODERS,
-                ),
+                )
+                // see https://github.com/gfx-rs/wgpu/issues/2521
+                .skip(FailureCase::always().flaky()),
```
Comment (cwfitzgerald) [inline examples/src/timestamp_queries/mod.rs]: Do we want to _skip_ or just mark as a flaky failure?

Comment (teoxoy) [reply examples/src/timestamp_queries/mod.rs]: I wanted to `skip` them at first then I saw we have a `flaky` expectation and forgot to change `skip` to `expect_fail`.
Will fix.

File: examples/src/timestamp_queries/mod.rs [hunk 2]
```diff
@@ -456,7 +456,9 @@ mod tests {
                 .features(
                     wgpu::Features::TIMESTAMP_QUERY
                         | wgpu::Features::TIMESTAMP_QUERY_INSIDE_ENCODERS,
-                ),
+                )
+                // see https://github.com/gfx-rs/wgpu/issues/2521
+                .expect_fail(FailureCase::always().flaky()),
```
Comment (cwfitzgerald) [inline examples/src/timestamp_queries/mod.rs]: Ack, sorry one more comment, can you add a message to make sure that other issues with the test are caught, `flaky` is a big hammer (ignore status of the test) so we want to refine it as much as possible.

Comment (teoxoy) [reply examples/src/timestamp_queries/mod.rs]: I added a "unexpected timestamp" panic message.

**Review by cwfitzgerald — APPROVED** [review]
Thanks!

**Comment by torokati44** [comment]
FYI auto-merge is blocked by the unresolved conversation.

**Pull request merged**